### DISCUSSION TITLE
Can moe work with metadata-less music?

### DISCUSSION THREAD

**Discussion by jaimet** [MoeMusic/Moe#270] [Q&A]
I'm trying to use the example script on the documentation homepage to tag an album that has no metadata whatsoever, but I'm getting the following error:

```
$ ../tag . $myRecordingId
Traceback (most recent call last):
  File "/home/user/moetest/music/../tag", line 33, in <module>
    main()
  File "/home/user/moetest/music/../tag", line 24, in main
    album = Album.from_dir(pathlib.Path(args.path))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/moetest/venv/lib/python3.11/site-packages/moe/library/album.py", line 396, in from_dir
    track = Track.from_file(file_path, album)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/moetest/venv/lib/python3.11/site-packages/moe/library/track.py", line 432, in from_file
    raise ValueError(
ValueError: Track is missing required fields. [missing_reqd_fields=['title', 'track_num', 'album_artist', 'album_title', 'date']]
```

Could someone please confirm that moe requires the above tags ("fields") to be already present in the (source) files in order to be able import them?

**Comment by jtpavlock** [comment]
That's correct, Moe cannot import music without those required fields present.

Edit: As a workaround, you should be able to just set dummy values for the required fields when you first create the Track if you use the default Track constructor.

**Comment by jaimet** [comment]
Sadly for me, that means that moe is a no-go, and as far as I can tell, it's not mentioned _anywhere_ in any of the documentation.
Would it be possible to add that info somewhere prominent?
 
Parenthetically, it's interesting that beets **can** perform this function. I think that it would be a great addition if, one day in the future, moe could also tag tracks from a known recording id despite having no pre-existing metadata.

**Reply by jtpavlock** [reply]
Thinking about this again, I believe the workaround would be quite simple. Instead of using the `from_file` constructor for tracks, you can use the default `__init__` for a Track and manually set dummy values for the required fields initially before merging in the proper fields from the musicbrainz track.
